src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -368,7 +368,7 @@ void java_bytecode_languaget::convert_lazy_method(
 /// generated by string_preprocess.
 /// \param context: a symbol table
 void java_bytecode_languaget::replace_string_methods(
-  symbol_tablet &context)
+  symbol_table_baset &context)
```

Comment:
Why do you generalise to the base class here, but not in the other methods?

Reply:
Daniel's instructions: minimise churn by only messing with those functions we have a use for at the moment. Since we're only observing symbol table changes callable from `::typecheck` and `languaget::final`, the result is some methods continuing to take a `symbol_tablet`. The first user, if one ever comes along, can take responsibility for generalising them.

Reply:
However, this PR should at least work with current TG. I'll list the necessary extensions below.